File: Reto # - 15/Python/Asincronia.py
import asyncio
## DeathbatO

async def ejecutar_tarea(nombre, segundos):
    print(f"Tarea '{nombre}' empezando.")
    print(f"Tarea '{nombre}' durará {segundos} segundo(s).")
    await asyncio.sleep(segundos)
    print(f"Tarea '{nombre}' finalizada.")

async def main():
    # Ejecutar las funciones C, B y A en paralelo
    tarea_c = ejecutar_tarea("C", 3)
    tarea_b = ejecutar_tarea("B", 2)
    tarea_a = ejecutar_tarea("A", 1)
    await asyncio.gather(tarea_c, tarea_b, tarea_a)

    # Ejecutar la función D después de que las funciones C, B y A han finalizado
    tarea_d = ejecutar_tarea("D", 1)
    await tarea_d

asyncio.run(main())

File: Reto # - 15/Java/Asincronia.java
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.Future;
import java.util.concurrent.TimeUnit;
//*
* @author DeathbatO
*/

class Asincronia implements Runnable {
    private String nombre;
    private int segundos;

    public Tarea(String nombre, int segundos) {
        this.nombre = nombre;
        this.segundos = segundos;
    }

    @Override
    public void run() {
        System.out.println("Tarea '" + nombre + "' empezando.");
        System.out.println("Tarea '" + nombre + "' durará " + segundos + " segundo(s).");
        try {
            TimeUnit.SECONDS.sleep(segundos);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        System.out.println("Tarea '" + nombre + "' finalizada.");
    }
}

public class Main {
    public static void main(String[] args) {
        ExecutorService executor = Executors.newFixedThreadPool(4);

        // Ejecutar las funciones C, B y A en paralelo
        Future<?> tareaC = executor.submit(new Tarea("C", 3));
        Future<?> tareaB = executor.submit(new Tarea("B", 2));
        Future<?> tareaA = executor.submit(new Tarea("A", 1));

        // Esperar a que las funciones C, B y A hayan finalizado
        try {
            tareaC.get();
            tareaB.get();
            tareaA.get();
        } catch (Exception e) {
            e.printStackTrace();
        }

        // Ejecutar la función D después de que las funciones C, B y A han finalizado
        executor.submit(new Tarea("D", 1));

        executor.shutdown();
    }
}
